src/tasks/task.ts

```diff
@@ -0,0 +1,145 @@
+import assign from "../util/assign";
+import { beacon } from "../util/beacon";
+import { getClientInfo } from "../util/client-info";
+import templateResource from "../lib/templateResource";
+import getNetworkInformation from "../lib/networkInformation";
+
+interface State {
+  hasRan: boolean;
+}
+// The VERSION var is replaced at compile time via webpack.DefinePlugin
+declare const VERSION: string;
+
+/* eslint-disable @typescript-eslint/camelcase */
+const blankBeacon: Beacon = {
+  client_user_agent: "",
+  client_ip: "",
+  client_asn: 0,
+  client_region: "",
+  client_country_code: "",
+  client_continent_code: "",
+  client_metro_code: "",
+  client_postal_code: "",
+  client_conn_speed: "",
+  client_gmt_offset: "",
+  client_latitude: "",
+  client_longitude: "",
+  resolver_ip: "",
+  resolver_asn: 0,
+  resolver_region: "",
+  resolver_country_code: "",
+  resolver_continent_code: "",
+  resolver_conn_speed: "",
+  resolver_latitude: "",
+  resolver_longitude: "",
+  test_id: "",
+  test_api_key: "",
+  test_lib_version: "",
+  test_server: "",
+  test_timestamp: 0,
+  task_type: "",
+  task_id: "",
+  task_schema_version: "",
+  task_client_data: "",
+  task_server_data: ""
```

Comment:
Should we add the Pop Id and Server ID from parsing the header to this beacon template?

Reply:
I think since this is a Fastly-specific format, it falls under what we were talking about earlier today, where we'd push beacon-handling back into the provider. I'll address this in the next changeset involving creation of the throughput methodology in the CBSi provider.